_format/1.1/index.md

```diff
@@ -1839,3 +1877,572 @@ An error object **MAY** have the following members:
+  operation to perform. Valid values include:
+  - `"get"`: fetches a resource, resource identity, collection of resources, or
+     collection of resource identitities
+  - `"add"`: creates a new resource or relationship
+  - `"update"`: updates a resource or relationship
+  - `"remove"`: removes a resource or relationship
+
+* `data`: the operation's "primary data".
+
+* `included`: an array of [resource
+  objects](#document-resource-objects) that are
+  related to the operation's primary data and/or each other
+  ("included resources").
+
+* `ref`: the resource or resource collection that is the target of the
+  operation. This member serves to disambiguate the target at a particular
+  endpoint if that endpoint represents multiple resources. `ref` **MUST** be
+  expressed as an object that **MAY** contain any of the following members:
+  - `type`: resource `type`
+  - `id` or `lid`: resource identifier or local identifier
+  - `relationship`: the name of the relationship
+
+* `params`: all of the query parameters allowed for singular requests can be
+  represented in JSON form in an operation's `params` object. Possible members
+  include:
+  - `include`
+  - `sort`
+  - `filter`
+  - `page`
+  - `fields`
+
+* `links`: a [links object](#document-links) related to the operation's primary
+  data.
+
+* `meta`: a [meta object](#document-meta) that contains non-standard
+  meta-information about the operation.
+
+Different members are required for processing different types of operations, as
+described below.
+
+> Note: `op` codes are used instead of the HTTP methods prescribed for singular
+  requests. The values `"get"`, `"add"`, `"update"`, and `"remove"` align with
+  the HTTP methods `GET`, `POST`, `PATCH`, and `DELETE` used for singular
+  requests.
+
+### <a href="#processing-operations" id="processing-operations" class="headerlink"></a> Processing Operations
+
+A server **MUST** perform operations in the order they appear in the
+`operations` array.
+
+All operations in a request **MUST** be performed atomically, so that a failure
+to perform any operation **MUST** invalidate any effects of preceding
+operations.
+
+A successful operation's `data` and `included` members **MUST** conform with the
+requirements for the analogous singular request and response.
+
+A server **MUST** respond to successful operations requests with `200 OK` if
+any operations are required to have `data` in a response. An array of resultant
+operation objects **MUST** be returned that correspond positionally to the
+requested operations.
+
+A server **MAY** respond to successful requests that include operations with
+`204 No Content` and no response document if no operations are required to
+return `data`.
+
+If any operation in a request fails, the server **MUST** respond as described
+in [Errors].
+
+> Note: An empty object (`{}`) is a valid operation object, and thus can be
+returned as a result to an operation request.
```

Comment:
Allowing an empty object to be a valid operations object makes sense in a response, but not a request, right? Should the spec make it clear that this is the case or am I missing something here?

Also, should this be more than a "Note"? This seems like an important part of the spec. The phrase "can be" in this note also seems out of place when we clearly use the imperatives everywhere else. Should it be `**MAY** be`?

Reply:
This is a note and not a normative portion of the spec because it does not introduce any new requirements or information. In other words, the fact that an empty object can be a valid operation object can be deduced from the description of the operation object. Since it's not normative, we shouldn't use language like `**MAY**`.